pkg/util/copyutil/clone_test.go

```diff
@@ -0,0 +1,86 @@
+package copyutil
+
+import (
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestClone(t *testing.T) {
+	Convey("it should clone a json", t, func() {
+		m1 := map[string]interface{}{
+			"a": "bbb",
+			"b": map[string]interface{}{
+				"c": 123,
+			},
+		}
+
+		m2, err := Clone(m1)
+		So(err, ShouldBeNil)
+		So(m2, ShouldResemble, m1)
+
+		// change m1
+		m1["a"] = "zzz"
+		delete(m1, "b")
```

Comment:
Try to change "c" to ensure nested objects are also deep cloned?

Reply:
Added [Ensure nested objects are also deep cloned in copyutil.Clone](https://github.com/authgear/authgear-server/pull/4729/commits/e1b6c0a397a31445f8e1e90d8532bd7d9a2719b5)